patches/openssl/0001-Add-managarm-support.patch

```diff
@@ -0,0 +1,101 @@
+From c679fd70dc11dbe227743ed0fb4ec02e624a7e04 Mon Sep 17 00:00:00 2001
+From: Dennis Bonke <admin@dennisbonke.com>
+Date: Sun, 21 Feb 2021 13:22:15 +0100
+Subject: [PATCH] Add managarm support
+
+Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
+---
+ apps/speed.c       |  2 ++
+ engines/e_afalg.c  | 12 +++++++++++-
+ test/afalgtest.c   |  7 +++++++
+ test/rsa_complex.c |  4 +++-
+ 4 files changed, 23 insertions(+), 2 deletions(-)
+
+diff --git a/apps/speed.c b/apps/speed.c
+index d4ae7ab7bf..db2db31bcd 100644
+--- a/apps/speed.c
++++ b/apps/speed.c
+@@ -113,6 +113,8 @@
+ # define NO_FORK
+ #endif
+ 
++#include <sys/select.h>
++
+ #define MAX_MISALIGNMENT 63
+ #define MAX_ECDH_SIZE   256
+ #define MISALIGN        64
+diff --git a/engines/e_afalg.c b/engines/e_afalg.c
+index 4b17228461..5a445e8154 100644
+--- a/engines/e_afalg.c
++++ b/engines/e_afalg.c
+@@ -21,6 +21,7 @@
+ #include "internal/nelem.h"
+ 
+ #include <sys/socket.h>
++#if !defined (__managarm__)
```

Comment:
What do other OSes do here? I.e., OSes that do not have AFALG?

Reply:
AFALG is Linux specific as far as I can see. So I guess they disable it as well

Reply:
Exactly, that's why I am asking how other OSes disable this. Maybe we should use the same strategy as the other OSes to disable this.

Reply:
I’ve checked haiku, and they don’t explicitly disable it, but they do pass no-asm which may affect it. I’m not aware of any other ports of OpenSSL outside of Linux, the BSD’s and haiku